### PR TITLE
Add dedicated documentation for shared links

### PR DESCRIPTION
**Pull request by cbetta**
We recently added dedicated API documentation for shared links.

https://developer.box.com/reference/get-shared-items/

To CRUD shared links we actually need to use the file/folder update endpoints, which was documented generically but not for this specific use-case.

By adding these docs, we can add code samples to the API endpoints for these specific use-cases.

### CODE REVIEW

File: doc/files.md [hunk 1]
```diff
@@ -631,17 +660,49 @@ BoxSharedLink sharedLink = file.createSharedLink(BoxSharedLink.Access.OPEN, null
 
 [create-shared-link]: http://opensource.box.com/box-java-sdk/javadoc/com/box/sdk/BoxFile.html#createSharedLink-com.box.sdk.BoxSharedLink.Access-java.util.Date-com.box.sdk.BoxSharedLink.Permissions-
 
+Get a Shared Link
+-----------------
+
+A shared link for a file can be generated by calling
```
Comment (PJSimon) [inline doc/files.md:666]:
```suggestion
Retrieve the shared link for a file by calling
```

File: doc/folders.md [hunk 2]
```diff
@@ -237,50 +241,96 @@ folder.delete(true);
...
+Date unshareDate = new Date();
+BoxSharedLink sharedLink = folder.createSharedLink(BoxSharedLink.Access.OPEN, null, permissions);
+```
+
+[create-shared-link]: http://opensource.box.com/box-java-sdk/javadoc/com/box/sdk/BoxFolder.html#createSharedLink-com.box.sdk.BoxSharedLink.Access-java.util.Date-com.box.sdk.BoxSharedLink.Permissions-
+
+Get a Shared Link
+-----------------
+
+A shared link for a folder can be generated by calling
```
Comment (PJSimon) [inline doc/folders.md:291]:
```suggestion
Retrieve the shared link for a folder by calling
```

File: doc/folders.md [hunk 3]
```diff
@@ -237,50 +241,96 @@ folder.delete(true);
...
 
 <!-- sample put_folders_id_shared_link_remove -->
 ```java
 BoxFolder folder = new BoxFolder(api, "12345");
-BoxFolder.Info info = folder.getInfo();
-info.removeSharedLink()
-folder.updateInfo(info)
+BoxFolder.Info info = file.getInfo();
+info.removeSharedLink();
+file.updateInfo(info);
```
Comment (PJSimon) [inline doc/folders.md:330]:
```suggestion
BoxFolder.Info info = folder.getInfo();
info.removeSharedLink();
folder.updateInfo(info);
```

**Comment by PJSimon** [comment]
@cbetta If my suggestions look good, you can batch them into one commit on the Files tab.

File: doc/files.md [hunk 4]
```diff
@@ -631,17 +660,49 @@ BoxSharedLink sharedLink = file.createSharedLink(BoxSharedLink.Access.OPEN, null
```
Comment (PJSimon) [inline doc/files.md:660]:
```suggestion

A set of shared link access level constants are available through the SDK for convenience:
* `BoxSharedLink.Access.OPEN`
* `BoxSharedLink.Access.COLLABORATORS`
* `BoxSharedLink.Access.COMPANY`
* `BoxSharedLink.Access.DEFAULT`
* `accessLevels.DISABLED`

```

File: doc/folders.md [hunk 5]
```diff
@@ -237,50 +241,96 @@ folder.delete(true);
...
-BoxFolder.Info info = folder.new Info();
-info.setSharedLink(sharedLink);
-folder.updateInfo(info);
+BoxSharedLink.Permissions permissions = new BoxSharedLink.Permissions();
+permissions.setCanDownload(true);
+permissions.setCanPreview(true);
+Date unshareDate = new Date();
+BoxSharedLink sharedLink = folder.createSharedLink(BoxSharedLink.Access.OPEN, null, permissions);
+```
+
```
Comment (PJSimon) [inline doc/folders.md:285]:
```suggestion

A set of shared link access level constants are available through the SDK for convenience:
* `BoxSharedLink.Access.OPEN`
* `BoxSharedLink.Access.COLLABORATORS`
* `BoxSharedLink.Access.COMPANY`
* `BoxSharedLink.Access.DEFAULT`
* `accessLevels.DISABLED`

```

**Comment by cbetta** [comment]
Thank you for the suggestions @PJSimon. All merged.

**Review by PJSimon — APPROVED** [review]

**Pull request merged**